xds/src/main/java/io/grpc/xds/XdsServerBuilder.java

```diff
@@ -130,9 +132,18 @@ public Server build() {
             filterChainSelectorManager, xdsClientPoolFactory, filterRegistry);
   }
 
-  @VisibleForTesting
-  XdsServerBuilder xdsClientPoolFactory(XdsClientPoolFactory xdsClientPoolFactory) {
-    this.xdsClientPoolFactory = checkNotNull(xdsClientPoolFactory, "xdsClientPoolFactory");
+  /**
+   * Allows injecting {@link XdsClientPoolFactory} and/or overriding bootstrap configuration, useful
+   * for testing.
+   */
+  public XdsServerBuilder xdsClientPoolFactory(@Nullable XdsClientPoolFactory xdsClientPoolFactory,
```

Comment:
XdsClientPoolFactory is not public, and we don't want it to be public. We just want the bootstrap. And we should probably put "for test" or the like in the name.